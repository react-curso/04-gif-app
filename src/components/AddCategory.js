import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ categories }) => { 

     const [inputValue, setInputValue] = useState('');

     const handleInputChange = (e) => {
          setInputValue(e.target.value);
     }

     const handleSubmit = (e) => {
          e.preventDefault();
          if (inputValue.trim().length > 2) {
               categories(c => [ inputValue, ...c]);
               setInputValue('');

               
          }
     }

     return (
          <>
               <form onSubmit={handleSubmit} >
                    <input
                         type="text"
                         value={inputValue}
                         onChange={handleInputChange}
                    />
               </form>
          </>
     )
}

AddCategory.propTypes = {
     categories: PropTypes.func.isRequired
}

