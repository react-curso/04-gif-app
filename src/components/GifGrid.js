import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            { loading && <p>Loading</p> }            

            <div className="card-grid">
                {
                    images.map((img) => (

                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}
