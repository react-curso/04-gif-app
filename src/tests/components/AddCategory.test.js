import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import { AddCategory } from '../../components/AddCategory'

describe('test component AddCategory', () => {
    
    const categories = jest.fn(); // simula la funcion con jest

    let wrapper = shallow(<AddCategory  categories={categories} />);

    beforeEach( () => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory  categories={categories} />);


    } )

    test('render component', () => {
     
        expect( wrapper ).toMatchSnapshot();

    })

    // test('change in input text', () => {
        
    //     const input = wrapper.find('input');

    //     const value = 'hola' 

    //     input.simulate('change', { target: { value } });


    // })

    // test('no postear la informacion con submit ', () => {
        
    //     // forma corta de definir una funcion preventDefault(){};
    //     wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    //     expect( categories ).not.toHaveBeenCalled();

    // })
    

    test('debe de llamar setcategories y limpiar la caja ', () => {
      
        
        const input = wrapper.find('input');
        const value = 'goku'

        input.simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault(){} } );
        
        expect( categories ).toHaveBeenCalled();
        expect( categories ).toHaveBeenCalledWith( expect.any(Function) ); // espera una funcion

        const valueInput = input.text();
        
        expect( valueInput ).toBe('');

    })
    
    
    

})
