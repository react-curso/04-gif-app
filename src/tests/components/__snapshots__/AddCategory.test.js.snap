// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component AddCategory render component 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
